[SPARK-25105][PySpark] [SQL]Include PandasUDFType in the import all of pyspark.sql.functions

kevinyu98 opened the pull request:
## What changes were proposed in this pull request?

Include PandasUDFType in the import all of pyspark.sql.functions 

## How was this patch tested?

Run the test case from the pyspark shell from the jira [spark-25105](https://jira.apache.org/jira/browse/SPARK-25105?jql=project%20%3D%20SPARK%20AND%20component%20in%20(ML%2C%20PySpark%2C%20SQL%2C%20%22Structured%20Streaming%22))
I manually test on pyspark-shell:
before:
`
>>> from pyspark.sql.functions import *
>>> foo = pandas_udf(lambda x: x, 'v int', PandasUDFType.GROUPED_MAP)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'PandasUDFType' is not defined
>>> 
`
after:
`
>>> from pyspark.sql.functions import *
>>> foo = pandas_udf(lambda x: x, 'v int', PandasUDFType.GROUPED_MAP)
>>> 
`
Please review http://spark.apache.org/contributing.html before opening a pull request.

kevinyu98 committed:
include PandasUDFType in the import all

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

HyukjinKwon commented:
ok to test

HyukjinKwon commented:
Mind linking the JIRA by fixing the title like `[SPARK-25105][PYTHON] Include ...`?

SparkQA commented:
**[Test build #94735 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/94735/testReport)** for PR 22100 at commit [`8201eaf`](https://github.com/apache/spark/commit/8201eafd6136d9a3998de7c1574b5aa20993c6c7).

SparkQA commented:
**[Test build #94735 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/94735/testReport)** for PR 22100 at commit [`8201eaf`](https://github.com/apache/spark/commit/8201eafd6136d9a3998de7c1574b5aa20993c6c7).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/94735/
Test FAILed.

kevinyu98 committed:
add one more space

dilipbiswal commented:
@kevinyu98 Can you please fix the description to "IncludeInclude PandasUDFType in the import all of pyspark.sql.functions" or anything better.

dilipbiswal commented on python/pyspark/sql/functions.py:
```diff
@@ -2927,7 +2927,7 @@ def pandas_udf(f=None, returnType=None, functionType=None):
            if not k.startswith('_') and k[0].islower() and callable(v) and k not in blacklist]
 __all__.sort()
-
+__all__ += ["PandasUDFType"]
```
@kevinyu98 should this be before the sort() ?

kevinyu98 replied:
yah, let me change it. Thanks

kevinyu98 committed:
reverse the change in tests.py

kevinyu98 committed:
move the append on top of sort

SparkQA commented:
**[Test build #94768 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/94768/testReport)** for PR 22100 at commit [`4c67e07`](https://github.com/apache/spark/commit/4c67e07d64818c83652b9050d5dd120a10cbac69).

kevinyu98 commented:
@HyukjinKwon @dilipbiswal Thanks for your comments, I have changed the code. I removed the code changes in tests.py, because it got warning message during style checking, and I did manually testing and add the result in the pr description.

`Qianyangs-MBP:spark qianyangyu$ ./dev/lint-python 
./python/pyspark/sql/tests.py:4357: SyntaxWarning: import * only allowed at module level
  @unittest.skipIf(
pycodestyle checks passed.
rm -rf _build/*
pydoc checks passed.`

kevinyu98 commented:
@holdenk Hello Holden: Can you check the code changes? Is this what you expected behavior? Thanks.

SparkQA commented:
**[Test build #94767 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/94767/testReport)** for PR 22100 at commit [`06e41b5`](https://github.com/apache/spark/commit/06e41b5ff60b45a95eb8712e645d909d1456349d).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/94767/
Test PASSed.

SparkQA commented:
**[Test build #94768 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/94768/testReport)** for PR 22100 at commit [`4c67e07`](https://github.com/apache/spark/commit/4c67e07d64818c83652b9050d5dd120a10cbac69).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/94768/
Test PASSed.

BryanCutler reviewed (approved):
LGTM
